modules/connectors/hay_post/karrio/providers/hay_post/shipment/create.py

```diff
@@ -0,0 +1,119 @@
+import typing
+import karrio.lib as lib
+import karrio.core.units as units
+import karrio.core.models as models
+import karrio.providers.hay_post.error as error
+import karrio.providers.hay_post.utils as provider_utils
+import karrio.providers.hay_post.units as provider_units
+import karrio.schemas.hay_post.order_create_request as hay_post
+import karrio.schemas.hay_post.order_create_response as hay_post_response
+
+
+def parse_shipment_response(
+    _response: lib.Deserializable[dict],
+    settings: provider_utils.Settings,
+) -> typing.Tuple[models.ShipmentDetails, typing.List[models.Message]]:
+    response = _response.deserialize()
+
+    messages = []
+    shipment = []
+    if response.get('key') is None:
+        shipment = _extract_details(response, settings)
+    else:
+        messages = error.parse_error_response(response, settings)
```

Comment:
The default value for shipment is `None` not `[]`.  That could cause an error upstream.

I would recommend updating the parse_error_response function to filter the response for errors so that it is centralized. meaning you can always send a response or list of responses and it should be able to check if there is an error and extract it.

```suggestion
    messages = error.parse_error_response(response, settings)
    shipment = (
        _extract_details(response, settings)
        if response.get('key') is not None
        else None
    )
```